CONTRIBUTORS

```diff
@@ -20,20 +29,30 @@ Thank you!
     Alex Dowad <alexinbeijing@gmail.com>
     Alexey Veselovsky <alexey.veselovsky@eykontech.com>
     Alexis Robert <alexis.robert@gmail.com>
+    Alex Rousskov
+    Alex Rousskov <>
+    Alex Rousskov <rousskov@measurement-factory.com
     Alex Rousskov <rousskov@measurement-factory.com>
     Alex Rousskov <rousskov@squid-cache.org>
     Alex Wu <alex_wu2012@hotmail.com>
     Alin Nastac <mrness@gentoo.org>
     Alter <alter@alter.org.ua>
+    Ambrose Li <ambrose.li@gmail.com>
+    Amish <3330468+amishmm@users.noreply.github.com>
+    Amish <amishxda@users.noreply.github.com>
     Amit Klein <amit.klein@safebreach.com>
+    amos@georgia.treenetnz.com <>
     Amos Jeffries
+    Amos Jeffries <>
+    amosjeffries <>
     Amos Jeffries <amosjeffries@squid-cache.org>
     Amos Jeffries <squid3@treenet.co.nz>
     Amos Jeffries <yadij@users.noreply.github.com>
     Amos <squid3@treenet.co.nz>
```

Comment:
You mean simply placing '#' comments in front of the entries not to be published and filtering them out in the release/snapshot scripts? If so, seems good to me.

Reply:
Folks wanting to hide their involvement are not relevant. "Anonymous" is already an author to give attribution to if someone sends you a patch and ask not to be mentioned. Others provide us with an alias from the outset and do not even let us know their real identity - which is fine too. The data this list is being generated from is already published and publicly available.

The problem to solve is entries which have outdated details or typos from the bzr days when I had to manually a--author everything accepted. eg the 5 entries for me and the 4 for you. The core dev team having old CVS surname-only labels as well as full name+email entries. Basically the author name+email it is too late to go back in history and fix.

Reply:
> Folks wanting to hide their involvement are not relevant

I disagree that participation in an open source project implies _mandatory permanent_ listing in `CONTRIBUTORS`, but we do not need to agree on this issue to make progress -- if you think we should not care about this hypothetical use case, my "deal with it later if needed" suggestion allows us to ignore it for now without further arguments.

> The problem to solve is entries which have outdated details or typos

I hope that my "give `CONTRIBUTORS` special duplicate resolution power" suggestion addresses all of these problems (that we care about). If (and only if) it does not, we will add entry editing capabilities similar to the entry deleting capabilities we just discussed (i.e. `@Edit` instead of `@Silence` and probably using regexes instead of hash keys).

I will experiment and share the results.

Reply:
> > Folks wanting to hide their involvement are not relevant
> 
> I disagree that participation in an open source project implies _mandatory permanent_ listing in `CONTRIBUTORS`, 

You misunderstand me.

CONTRIBUTORS is a convenience copy of the " _mandatory permanent_ " VCS history. AFAIK it has two purposes, a) to highlight the appreciation for those contributors and b) for legal analysis of copyrights.

Folks that want to hide their involvement make arrangements to be formally attributed as "Anonymous". So it is clear to those legal folks that the author is *not* you or me etc. We have "Anonymous" in the CONTRIBUTORS already from past occurances of this case. So IMO these folks situation is not relevant to this PR and discussion.


> > The problem to solve is entries which have outdated details or typos
> 
> I agree that this problem exists and should be solved. I hope that my "give `CONTRIBUTORS` special duplicate resolution power" suggestion addresses all of the corresponding problematic cases (that we care about). If (and only if) it does not, we will add entry editing capabilities similar to the entry deleting capabilities we just discussed (i.e. `@Edit` instead of `@Silence` and probably using regexes instead of hash keys).
> 
> I will experiment and share the results.

I occurs to me we might hard-code a commit ID into the script and only auto-include the authors since that commit. Then we could manually filter out the bad entries from CVS/BZR days.

Reply:
Resolving since a better idea mentioned in the change request has been implemented. This change request is not the right place to try to agree regarding CONTRIBUTORS effect on copyright and other out-of-scope issues.